agent/agent-profiler/agent-alerting-api/src/main/java/com/microsoft/applicationinsights/alerting/alert/AlertBreach.java

```diff
@@ -80,7 +80,9 @@ public static AlertBreach.Builder builder() {
   @Override
   public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
     jsonWriter.writeStartObject();
-    jsonWriter.writeStringField("type", type.name());
+    if (type != null) {
```

Comment:
@johnoliver is type expected to be null? do i need to do null check for other enum types everywhere? please help me confirm.

Reply:
it would probably be preferable to have absent rather than null, I would do the null check